bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -822,10 +831,10 @@ public void run()
       };
     }
 
-    protected void release(boolean wait)
+    protected void release(boolean wait, boolean writer)
     {
       final int refCount = this.refCount.decrementAndGet();
-      if (refCount == 0 && storage != null) {
+      if (refCount == 0 && storage != null && canRelease(writer)) {
```

Comment:
@vrozov I think you make some good points but those could be covered by separate attributes like number of blocks when to start apply pressure or as sandesh suggested a pluggable policy and number of blocks being a property of one such policy. In fact, I might be able to add the ability to configure number of blocks (taking spooling into account) trivially.